1-js/02-first-steps/05-types/article.md

```diff
@@ -1,226 +1,250 @@
+`undefined` նշանակում է, որ «արժեք չի վերագրվել»։
 
-If a variable is declared, but not assigned, then its value is `undefined`:
+Եթե փոփոխականը հայտարարված է, բայց նրան արժեք չի վերագրվել, ապա նրա արժեքը կլինի `undefined`․
 
 ```js run
 let age;
 
-alert(age); // shows "undefined"
+alert(age); // ցույց կտա «undefined»
 ```
 
-Technically, it is possible to explicitly assign `undefined` to a variable:
+Տեխնիկապես փոփոխականին հնարավոր է ակնհայտ կերպով վերագրել `undefined` արժեքը․
 
 ```js run
 let age = 100;
 
-// change the value to undefined
+// արժեքը դարձնել «undefined»
 age = undefined;
 
-alert(age); // "undefined"
+alert(age); // «undefined»
 ```
 
-...But we don't recommend doing that. Normally, one uses `null` to assign an "empty" or "unknown" value to a variable, while `undefined` is reserved as a default initial value for unassigned things.
+...Բայց խորհուրդ չի տրվում այդպես անել։ Սովորաբար `null`-ը օգտագործվում է փոփոխականին «դատարկ» կամ «անհայտ» արժեք վերագրելու համար, 
+մինչդեռ `undefined`-ը  իրենից ենթադրում է նախնական լռելյայն արժեք՝ չվերագրված փոփոխականների համար։
 
-## Objects and Symbols
+## Օբյեկտներ և սիմվոլներ
 
-The `object` type is special.
+`object` տիպը յուրահատուկ է։
 
-All other types are called "primitive" because their values can contain only a single thing (be it a string or a number or whatever). In contrast, objects are used to store collections of data and more complex entities.
+Մնացած բոլոր տիպերը կոչվում են «պրիմիտիվ» տիպեր, քանի որ նրանց արժեքները պարզ արժեքներ են (լինի այն տող, թիվ կամ այլ արժեք)։ Օբյեկտները, 
+իրենց հերթին, օգտագործվում են տվյալների համախումբ և ավելի բարդ կառուցվածքներ պահեստավորելու նպատակով։
 
-Being that important, objects deserve a special treatment. We'll deal with them later in the chapter <info:object>, after we learn more about primitives.
+Օբյեկտները լեզվում կարևոր տեղ են զբաղեցնում և հատուկ ուշադրության են արժանի։ Մենք կզբաղվենք դրանցով ավելի ուշ՝ <info:object> գլխում՝ 
+պրիմիտիվ տիպերը ավելի խորը ուսումնասիրելուց հետո։
 
-The `symbol` type is used to create unique identifiers for objects. We have to mention it here for the sake of completeness, but also postpone the details till we know objects.
+`symbol` տիպը օգտագործվում է օբյեկտներում չկրկնվող նույնացուցիչներ ստեղծելու համար։ Մենք այստեղ նշում ենք դրա գոյության փաստը՝ տիպերի 
+նկարագիրը լիարժեք դարձնելու համար, բայց թողնում ենք դրա ուսումնասիրությունը հետագայի համար, երբ արդեն ուսումնասիրած կլինենք օբյեկտները։
 
-## The typeof operator [#type-typeof]
+## typeof օպերատորը [#type-typeof]
 
-The `typeof` operator returns the type of the argument. It's useful when we want to process values of different types differently or just want to do a quick check.
+`typeof` օպերատորը վերադարձնում է իր արգումենտի տիպը։ Այն օգտակար է, երբ մենք ուզում ենք մշակել տարբեր տիպերի արժեքները տարբեր կերպով, կամ 
+պարզեպես արագ ստուգման կարիք կա։
 
-It supports two forms of syntax:
+Այն ունի օգտագործման երկու սինտաքս․
```

Comment:
Instead of the word `սինտաքս` I'd prefer to actually use a translation. I usually use Microsoft Terminology Translator in case I am in doubt for guidance. https://www.microsoft.com/en-us/language/Search?&searchTerm=syntax&langID=49&Source=true&productid=0

Reply:
In case I feel that the translated term feels too alien I also supply the original English word in parentheses for reference and then just use the translated term afterwards.